### DISCUSSION TITLE
Does Ultragrid support multiple streams on the same machine?

### DISCUSSION THREAD

**Discussion by ygreq** [CESNET/UltraGrid#144] [Q&A]
Hi there,

Does Ultragrid support sending and receiving AV data to and from more than one other device? For example one pc to communicate at the same time with 2 other pcs, sending and receiving AV data to and from them?

Otherwise, is broadcasting possible?
If yes, may I ask how?

Thank you so much,
ygreq

**Comment by alatteri** [comment]
using the reflector you can take a single stream and send it to multiple recievers.

**Reply by ygreq** [reply]
Thank you, Alan! Would you mind elaborating a bit?  Maybe with a command. This is the first time I hear about reflector. 
BTW, do you know where I can find more info about each command? I read the pdf from janet and I also know how to use the help command, but the info I sometimes get is not really understandable. :/

Thank you so much!

**Reply by MartinPulec** [reply]
You can start with [this page](https://github.com/CESNET/UltraGrid/wiki/Reflector-settings). I am just not completely sure from the original post if it is what you need? Reflectors can be used for 1-to-N transmissions (usually unidirectional broadcast). If you need only 2 (or more) P2P transmissions from within one computer, then just use a separate port number (the `-P` parameter) for different streams (default UG port is UDP/5004 for video and UDP/5006 audio). Just remember that there should be at least offset of 4 between the parameters given to `-P`.

To sum up, Alan's answer is completely correct - the reflector can be used for one stream to more recipients. Using different ports can be used for multiple streams with more peers (perhaps bidirectional), also a difference is whether you want to use one capture device for all or a different one for any peer.

**Comment by alatteri** [comment]
https://github.com/CESNET/UltraGrid/wiki

https://github.com/CESNET/UltraGrid/wiki/Reflector-settings
